scipy/stats/_continuous_distns.py

```diff
@@ -7040,11 +7040,24 @@ def _stats(self, df):
     def _entropy(self, df):
         if df == np.inf:
             return norm._entropy()
-        half = df/2
-        half1 = (df + 1)/2
-        return (half1*(sc.digamma(half1) - sc.digamma(half))
-                + np.log(np.sqrt(df)*sc.beta(half, 0.5)))
 
+        def regular(df):
+            half = df/2
+            half1 = (df + 1)/2
+            return (half1*(sc.digamma(half1) - sc.digamma(half))
+                    + np.log(np.sqrt(df)*sc.beta(half, 0.5)))
+
+        def asymptotic(df):
+            # Formula from Wolfram Alpha:
+            # "asymptotic expansion (d+1)/2 * (digamma((d+1)/2) - digamma(d/2)) 
+            #  + log(sqrt(d) * beta(d/2, 1/2))"
+            df = np.array(df, dtype=np.float64)
+            h = (norm._entropy() + 1./df + 1. / 4 * df**-2 - 1. / 6. * df**-3
+                 - 1. / 8. * df**-4 + 3. / 10. * df**-5 + 1 / 4 * df**-6)
```

Comment:
I'd prefer to avoid conversion to an array within a low-level function that is wrapped by `np.vectorize`. 
```suggestion
            h = (norm._entropy() + 1/df + (df**-2.)/4 - (df**-3.)/6
                 - (df**-4.)/8 + 3/10*(df**-5.) + (df**-6.)/4)
```